docs/library.md

```diff
@@ -60,3 +68,80 @@ Arguments include:
+const result = await withPostGraphQLContext(
+  {
+    pgPool: myPgPool,
+    jwtToken: '...',
+    jwtSecret: '...',
+    pgDefaultRole: '...',
+  },
+  async context => {
+    // You execute your GraphQL query in this function with the provided `context` object.
+    // The `context` object will not work for a GraphQL execution outside of this function.
+    return await graphql(
+      myPostGraphQLSchema, // This is the schema we created with `createPostGraphQLSchema`.
+      query,
+      null,
+      { ...context }, // Here we use the `context` object that gets passed to this callback.
+      variables,
+      operationName,
+    )
+  },
+)
+```
+
+The exact APIs for `createPostGraphQLSchema` and `withPostGraphQLContext` are as follows.
+
+### `createPostGraphQLSchema(pgConfig?, schemaName? = 'public', options?): Promise<GraphQLSchema>`
+
+Arguments include:
+
+- **`pgConfig`**: An object or string that will be passed to the [`pg`][] library and used to connect to a PostgreSQL backend. If you already have a client or pool instance, when using this function you may also pass a `pg` client or a `pg-pool` instance directly instead of a config.
+- **`schemaName`**: A string which specifies the PostgreSQL schema that PostGraphQL will use to create a GraphQL schema. The default schema is the `public` schema. May be an array for multiple schemas. For users who want to run the Postgres introspection query ahead of time, you may also pass in a `PgCatalog` instance directly.
+- **`options`**: An object containing other miscellaneous options. Most options are shared with the `postgraphql` middleware function. Options could be:
+  - `classicIds`: Enables classic ids for Relay 1 support. Instead of using the field name `__id` for globally unique ids, PostGraphQL will instead use the field name `id` for its globally unique ids. This means that table `id` columns will also get renamed to `rowId`.
+  - `dynamicJson`: Setting this to `true` enables dynamic JSON which will allow you to use any JSON as input and get any arbitrary JSON as output. By default JSON types are just a JSON string.
+  - `disableDefaultMutations`: Setting this to `true` will prevent the creation of the default mutation types & fields. Database mutation will only be possible through Postgres functions.
+  - `jwtPgTypeIdentifier`: The Postgres type identifier for the compound type which will be signed as a JWT token if ever found as the return type of a procedure. Can be of the form: `my_schema.my_type`. You may use quotes as needed: `"my-special-schema".my_type`.
+
+### `withPostGraphQLContext(options, callback): Promise<GraphQLExecutionResult>`
+
+With this function you are required to provide some options and a callback. The callback is expected to return a GraphQL execution result. Whatever value is returned by the callback will then be returned by `withPostGraphQLContext`. The context you get as an argument to `callback` will be invalid anywhere outside of the `callback` function.
```

Comment:
This should talk about Promises. How about something like:

This function sets up a PostGraphQL context, calls (and resolves) the callback function within this context, and then tears the context back down again finally resolving to the result of your function. The callback is expected to return a promise which resolves to a GraphQL execution result. The context you get as an argument to `callback` will be invalid anywhere outside of the `callback` function.